addons/website_profile/views/website_profile.xml

```diff
@@ -283,16 +283,18 @@
         <t t-if="user.badge_ids">
             <div class="row">
                 <div class="col-lg-3 o_user_badges" t-foreach="user.badge_ids" t-as="badge">
-                    <div class="card-body o_wprofile_bb_grey">
-                        <table><tbody><tr>
-                            <th>
-                                <img class="o_wprofile_img_thumb" t-attf-src="/web/image/gamification.badge/#{badge.badge_id.id}/image_small"/>
-                            </th>
-                            <th style="">
-                                <span class="font-weight-bold" t-field="badge.badge_id.name"/>
-                            </th>
-                        </tr></tbody></table>
-                    </div>
+                    <t t-if="badge.badge_id.website_published">
+                        <div class="card-body o_wprofile_bb_grey">
+                            <table><tbody><tr>
+                                <th>
+                                    <img class="o_wprofile_img_thumb" t-attf-src="/web/image/gamification.badge/#{badge.badge_id.id}/image_small"/>
+                                </th>
+                                <th style="">
+                                    <span class="font-weight-bold" t-field="badge.badge_id.name"/>
+                                </th>
+                            </tr></tbody></table>
+                        </div>
+                    </t>
```

Comment:
this should be fixed on saas-12.2 with 3aa4c2476041626fa11d800955fe353ff6249589. I just included it for testing purpose